nimare/tests/test_estimator_performance.py

```diff
@@ -0,0 +1,149 @@
+import numpy as np
+
+from ..correct import FDRCorrector, FWECorrector
+from ..meta import ale, kernel, mkda
+from ..meta.utils import compute_ma, get_ale_kernel
+from ..transforms import mm2vox
+
+
+@pytest.mark.parametrize(
+    "meta_alg",
+    [
+        pytest.param(ale.ALE, id="ale"),
+        pytest.param(mkda.MKDADensity, id="mkda"),
+        pytest.param(mkda.KDA, id="kda"),
+    ],
+)
+@pytest.mark.parametrize(
+    "kern",
+    [
+        pytest.param(kernel.ALEKernel(), id="ale_kernel"),
+        pytest.param(kernel.MKDAKernel(), id="mkda_kernel"),
+        pytest.param(kernel.KDAKernel(), id="kda_kernel"),
+        pytest.param(kernel.Peaks2MapsKernel(), id="p2m_kernel"),
+    ],
+)
+@pytest.mark.parametrize(
+    "corr",
+    [
+        pytest.param(FWECorrector(method="bonferroni"), id="fwe_bonferroni"),
+        pytest.param(
+            FWECorrector(method="montecarlo", voxel_thresh=0.001, n_iters=10, n_cores=1),
+            id="fwe_montecarlo",
+        ),
+        pytest.param(FDRCorrector(method="indep", alpha=0.05), id="fdr_indep"),
+        pytest.param(FDRCorrector(method="negcorr", alpha=0.05), id="fdr_negcorr"),
+    ],
+)
+def test_estimators(simulatedata_cbma, meta_alg, kern, corr, mni_mask):
+
+    # set up testing dataset
+    ground_truth_foci, dataset = simulatedata_cbma
+
+    # create meta-analysis
+    meta = meta_alg(kern)
+
+    # peaks2MapsKernel does not work with any meta-analysis estimator
+    if isinstance(kern, kernel.Peaks2MapsKernel):
+        # AttributeError: 'DataFrame' object has no attribute 'masker'
+        meta_expectation = pytest.raises(AttributeError)
+    elif isinstance(kern, kernel.KDAKernel) and isinstance(meta, ale.ALE):
+        # IndexError: index 20000 is out of bounds for axis 0 with size 10010
+        meta_expectation = pytest.raises(IndexError)
+    else:
+        meta_expectation = does_not_raise()
+
+    with meta_expectation:
+        res = meta.fit(dataset)
+    # if creating the result failed (expected), do not continue
+    if isinstance(meta_expectation, type(pytest.raises(ValueError))):
+        return 0
+
+    # all combinations of meta-analysis estimators and multiple comparison correctors
+    # that do not work together
+    if isinstance(meta, mkda.MKDADensity) and isinstance(corr, FDRCorrector):
+        corr_expectation = pytest.raises(ValueError)
+    elif isinstance(meta, mkda.KDA) and isinstance(corr, FDRCorrector):
+        corr_expectation = pytest.raises(ValueError)
+    elif isinstance(meta, mkda.MKDADensity) and corr.method == "bonferroni":
+        corr_expectation = pytest.raises(ValueError)
+    elif isinstance(meta, mkda.KDA) and corr.method == "montecarlo":
+        # TypeError: correct_fwe_montecarlo() got an unexpected keyword argument 'voxel_thresh'
+        corr_expectation = pytest.raises(TypeError)
+    elif isinstance(meta, mkda.KDA) and corr.method == "bonferroni":
+        corr_expectation = pytest.raises(ValueError)
```

Comment:
@jdkent I think these should all be considered bugs. Would you be willing to open a separate issue for each of them? Preferably with the full tracebacks.

Reply:
didn't see the full traceback part, but the majority of errors come back [to here](https://github.com/neurostuff/NiMARE/blob/58e6e82e3ca15ae89390bbf4c294b44f689bd5a0/nimare/correct.py#L64-L69).